### DISCUSSION TITLE
4.2 Geopandas Join - Challenges with using Codespace .py

### DISCUSSION THREAD

**Discussion by travisz09** [ua-gist604b-f24/syllabus#14] [General]
I know that you are still working on the lectures for this module, and that I am working ahead. So I apologize if you are already intending on addressing this issue.

While I was able to complete the instructions for module 4.2, I had some difficulty in seeing my rendered outputs in Codespace. Code snipets such as `nybb.head()` would not render in the console without being passed to a print() statement, and none of the plots would actually show in the Codespace environment. From what I have [read](https://stackoverflow.com/questions/74415793/matplotlib-figures-not-generating-in-github-codespaces) online, Codespaces does not seem to support plot windows in the same way that a local ide would. The suggestion I found was to convert the .py file to a jupyter notebook (.ipynb), which supports more types of outputs, including plots.

The instructions clearly say to use a .py extension. I assume that this is so that you can download the code and run it locally. I was able to get my code to run without errors, however I was working blind most of the time. I could use print() statements to check on some things, but I was never able to figure out how to get a plot to render.

Is there something that I am missing? Should we convert this script into a Jupyter notebook?

I will hold off on pulling my assignment branch until this is cleared up.

Thank you,
Travis

**Comment by travisz09** [comment]
This did make a lot more sense after watching the lecture. Much of the data exploration code, such as print() and plot() is only for our benefit while assembling the spatial_join.py script, and is not required as part of the script itself. This was not clear to me from the assignment text, but I now see my mistake.

Apologies,
Travis

**Reply by aaryno** [reply]
Yep. If you want to be clever you can write the plots to files.
